diff -Nur gawk-4.0.1/awkgram.c gawk-4.0.1.neu/awkgram.c
--- gawk-4.0.1/awkgram.c	2012-03-28 22:03:33.000000000 +0200
+++ gawk-4.0.1.neu/awkgram.c	2012-05-31 23:11:07.573543685 +0200
@@ -4519,6 +4519,13 @@
 {"asort",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1)|A(2)|A(3),	do_asort},
 {"asorti",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1)|A(2)|A(3),	do_asorti},
 {"atan2",	Op_builtin,	 LEX_BUILTIN,	NOT_OLD|A(2),	do_atan2},
+{"bank_bic",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_bic},
+{"bank_name",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_name},
+{"bank_name_kurz",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_name_kurz},
+{"bank_ort",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_ort},
+{"bank_plz",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_plz},
+{"bank_pz",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_pz},
+{"bank_valid",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_valid},
 {"bindtextdomain",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1)|A(2),	do_bindtextdomain},
 {"break",	Op_K_break,	 LEX_BREAK,	0,		0},
 {"case",	Op_K_case,	 LEX_CASE,	GAWKX,		0},
@@ -4543,15 +4550,23 @@
 {"gensub",	Op_sub_builtin,	 LEX_BUILTIN,	GAWKX|A(3)|A(4), 0},
 {"getline",	Op_K_getline_redir,	 LEX_GETLINE,	NOT_OLD,	0},
 {"gsub",	Op_sub_builtin,	 LEX_BUILTIN,	NOT_OLD|A(2)|A(3), 0},
+{"iban_check",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_iban_check},
+{"iban_gen",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(2), do_iban_gen},
 {"if",		Op_K_if,	 LEX_IF,	0,		0},
 {"in",		Op_symbol,	 LEX_IN,	0,		0},
 {"include",  Op_symbol,	 LEX_INCLUDE,	GAWKX,	0},
 {"index",	Op_builtin,	 LEX_BUILTIN,	A(2),		do_index},
 {"int",		Op_builtin,	 LEX_BUILTIN,	A(1),		do_int},
+{"ipi_check",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_ipi_check},
+{"ipi_gen",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_ipi_gen},
 {"isarray",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1),	do_isarray},
+{"konto_check",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(2), do_konto_check},
+{"kto_check_encoding",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_kto_check_encoding},
 {"length",	Op_builtin,	 LEX_LENGTH,	A(0)|A(1),	do_length},
 {"log",		Op_builtin,	 LEX_BUILTIN,	A(1),		do_log},
 {"lshift",	Op_builtin,    LEX_BUILTIN,	GAWKX|A(2),	do_lshift},
+{"lut_free",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(0), do_lut_free},
+{"lut_init",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(2), do_lut_init},
 {"match",	Op_builtin,	 LEX_BUILTIN,	NOT_OLD|A(2)|A(3), do_match},
 {"mktime",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1),	do_mktime},
 {"next",	Op_K_next,	 LEX_NEXT,	0,		0},
@@ -4562,6 +4577,8 @@
 {"printf",	Op_K_printf,	 LEX_PRINTF,	0,		0},
 {"rand",	Op_builtin,	 LEX_BUILTIN,	NOT_OLD|A(0),	do_rand},
 {"return",	Op_K_return,	 LEX_RETURN,	NOT_OLD,	0},
+{"retval2txt",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_retval2txt},
+{"retval2utf8",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_retval2utf8},
 {"rshift",	Op_builtin,    LEX_BUILTIN,	GAWKX|A(2),	do_rshift},
 {"sin",		Op_builtin,	 LEX_BUILTIN,	NOT_OLD|A(1),	do_sin},
 {"split",	Op_builtin,	 LEX_BUILTIN,	A(2)|A(3)|A(4),	do_split},
diff -Nur gawk-4.0.1/awkgram.y gawk-4.0.1.neu/awkgram.y
--- gawk-4.0.1/awkgram.y	2012-03-28 21:52:40.000000000 +0200
+++ gawk-4.0.1.neu/awkgram.y	2012-05-31 23:11:02.157295051 +0200
@@ -1829,6 +1829,13 @@
 {"asort",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1)|A(2)|A(3),	do_asort},
 {"asorti",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1)|A(2)|A(3),	do_asorti},
 {"atan2",	Op_builtin,	 LEX_BUILTIN,	NOT_OLD|A(2),	do_atan2},
+{"bank_bic",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_bic},
+{"bank_name",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_name},
+{"bank_name_kurz",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_name_kurz},
+{"bank_ort",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_ort},
+{"bank_plz",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_plz},
+{"bank_pz",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_pz},
+{"bank_valid",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_bank_valid},
 {"bindtextdomain",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1)|A(2),	do_bindtextdomain},
 {"break",	Op_K_break,	 LEX_BREAK,	0,		0},
 {"case",	Op_K_case,	 LEX_CASE,	GAWKX,		0},
@@ -1853,15 +1860,23 @@
 {"gensub",	Op_sub_builtin,	 LEX_BUILTIN,	GAWKX|A(3)|A(4), 0},
 {"getline",	Op_K_getline_redir,	 LEX_GETLINE,	NOT_OLD,	0},
 {"gsub",	Op_sub_builtin,	 LEX_BUILTIN,	NOT_OLD|A(2)|A(3), 0},
+{"iban_check",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_iban_check},
+{"iban_gen",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(2), do_iban_gen},
 {"if",		Op_K_if,	 LEX_IF,	0,		0},
 {"in",		Op_symbol,	 LEX_IN,	0,		0},
 {"include",  Op_symbol,	 LEX_INCLUDE,	GAWKX,	0},
 {"index",	Op_builtin,	 LEX_BUILTIN,	A(2),		do_index},
 {"int",		Op_builtin,	 LEX_BUILTIN,	A(1),		do_int},
+{"ipi_check",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_ipi_check},
+{"ipi_gen",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_ipi_gen},
 {"isarray",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1),	do_isarray},
+{"konto_check",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(2), do_konto_check},
+{"kto_check_encoding",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_kto_check_encoding},
 {"length",	Op_builtin,	 LEX_LENGTH,	A(0)|A(1),	do_length},
 {"log",		Op_builtin,	 LEX_BUILTIN,	A(1),		do_log},
 {"lshift",	Op_builtin,    LEX_BUILTIN,	GAWKX|A(2),	do_lshift},
+{"lut_free",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(0), do_lut_free},
+{"lut_init",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(2), do_lut_init},
 {"match",	Op_builtin,	 LEX_BUILTIN,	NOT_OLD|A(2)|A(3), do_match},
 {"mktime",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1),	do_mktime},
 {"next",	Op_K_next,	 LEX_NEXT,	0,		0},
@@ -1872,6 +1887,8 @@
 {"printf",	Op_K_printf,	 LEX_PRINTF,	0,		0},
 {"rand",	Op_builtin,	 LEX_BUILTIN,	NOT_OLD|A(0),	do_rand},
 {"return",	Op_K_return,	 LEX_RETURN,	NOT_OLD,	0},
+{"retval2txt",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_retval2txt},
+{"retval2utf8",	Op_builtin,	 LEX_BUILTIN,	GAWKX|A(1), do_retval2utf8},
 {"rshift",	Op_builtin,    LEX_BUILTIN,	GAWKX|A(2),	do_rshift},
 {"sin",		Op_builtin,	 LEX_BUILTIN,	NOT_OLD|A(1),	do_sin},
 {"split",	Op_builtin,	 LEX_BUILTIN,	A(2)|A(3)|A(4),	do_split},
diff -Nur gawk-4.0.1/awk.h gawk-4.0.1.neu/awk.h
--- gawk-4.0.1/awk.h	2012-03-28 21:47:53.000000000 +0200
+++ gawk-4.0.1.neu/awk.h	2012-05-31 23:11:03.244347274 +0200
@@ -1173,6 +1173,26 @@
 extern void free_context(AWK_CONTEXT *ctxt, int );
 extern void append_symbol(char *name);
 
+
+/* konto_check_awk.c */
+extern NODE *do_iban_check (int nargs);
+extern NODE *do_iban_gen (int nargs);
+extern NODE *do_ipi_gen (int nargs);
+extern NODE *do_ipi_check (int nargs);
+extern NODE *do_lut_init (int nargs);
+extern NODE *do_lut_free (int nargs);
+extern NODE *do_konto_check (int nargs);
+extern NODE *do_bank_valid (int nargs);
+extern NODE *do_bank_name (int nargs);
+extern NODE *do_bank_name_kurz (int nargs);
+extern NODE *do_bank_bic (int nargs);
+extern NODE *do_bank_plz (int nargs);
+extern NODE *do_bank_ort (int nargs);
+extern NODE *do_bank_pz (int nargs);
+extern NODE *do_kto_check_encoding (int nargs);
+extern NODE *do_retval2txt (int nargs);
+extern NODE *do_retval2utf8 (int nargs);
+
 /* builtin.c */
 extern double double_to_int(double d);
 extern NODE *do_exp(int nargs);
diff -Nur gawk-4.0.1/konto_check_awk.c gawk-4.0.1.neu/konto_check_awk.c
--- gawk-4.0.1/konto_check_awk.c	1970-01-01 01:00:00.000000000 +0100
+++ gawk-4.0.1.neu/konto_check_awk.c	2012-05-31 23:48:47.743350836 +0200
@@ -0,0 +1,439 @@
+/*
+ * konto_check_awk.c - Builtin functions for konto_check
+ */
+
+/* 
+ * Copyright (C) 1986, 1988, 1989, 1991-2010 the Free Software Foundation, Inc.
+ * Copyright (C) 2011 Michael Plugge <m.plugge@hs-mannheim.de> (for this file)
+ * 
+ * This file is part of GAWK, the GNU implementation of the
+ * AWK Programming Language.
+ * 
+ * GAWK is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * GAWK is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+ */
+
+
+#include "awk.h"
+#include "konto_check.h"
+
+#define POP_TWO_SCALARS(s1, s2) \
+s2 = POP_SCALAR(); \
+s1 = POP_SCALAR();
+
+/* do_lut_init - Initialisierung von konto_check.
+ * Als Parameter sind nur der LUT-Name und der Level vorgesehen.
+ */
+NODE *
+do_lut_init(int nargs)
+{
+	NODE *tmp,*t1,*t2;
+   char lutfile[512];
+   int ret,level;
+
+   if(nargs>1){
+      t2=POP_SCALAR();
+      level=(int)force_number(t2);
+      DEREF(t2);
+   }
+   else
+      level=DEFAULT_INIT_LEVEL;
+   if(nargs>0){
+      t1=POP_SCALAR();
+      force_string(t1);
+      if(t1->stlen<512){
+         strncpy(lutfile,t1->stptr,t1->stlen);
+         lutfile[t1->stlen]=0;
+      }
+      else{
+         strncpy(lutfile,t1->stptr,511);
+         lutfile[511]=0;
+      }
+      DEREF(t1);
+   }
+   else
+      *lutfile=0;
+
+   ret=lut_init(lutfile,level,0);
+	return make_number((AWKNUM)(ret));
+}
+
+/* do_lut_free - allen allokierten Speicher freigeben */
+NODE *
+do_lut_free(int nargs)
+{
+	return make_number((AWKNUM)(lut_cleanup()));
+}
+
+/*  do_iban_gen: aus BLZ und Kontonummer einen IBAN generieren */
+NODE *
+do_iban_gen(int nargs)
+{
+	NODE *t1,*t2;
+   char blz[16],kto[16],iban[128],*papier,*ptr,*dptr;
+   int ret;
+
+   POP_TWO_SCALARS(t1,t2);
+   force_string(t1);
+   force_string(t2);
+   if(t1->stlen>8){
+      DEREF(t1);
+      DEREF(t2);
+      return make_string("",0);
+   }
+   if(t2->stlen>15){
+      DEREF(t1);
+      DEREF(t2);
+      return make_string("",0);
+   }
+   strncpy(blz,t1->stptr,15);
+   blz[t1->stlen]=0;
+   strncpy(kto,t2->stptr,15);
+   kto[t2->stlen]=0;
+   DEREF(t1);
+   DEREF(t2);
+   papier=iban_gen(blz,kto,&ret);
+   if(ret==LUT2_NOT_INITIALIZED){
+      lut_init(NULL,0,0);
+      papier=iban_gen(blz,kto,&ret);
+   }
+   if(ret<=0)return make_string("",0);
+
+      /* normale Version (nicht Papier) der IBAN erzeugen */
+   for(ptr=papier,dptr=iban;*ptr;ptr++)if(*ptr!=' ')*dptr++=*ptr;
+   *dptr=0;
+   kc_free(papier);
+	return make_string(iban,strlen(iban));
+}
+
+/*  do_ipi_check: Strukturierten Verwendungszweck (International Payment Instruction, IPI) testen */
+NODE *
+do_ipi_check(int nargs)
+{
+	NODE *t1,*t2;
+   char ipi[256];
+   int ret;
+
+   t1=POP_SCALAR();
+   force_string(t1);
+   if(t1->stlen>255){
+      DEREF(t1);
+      return make_string("",0);
+   }
+   strncpy(ipi,t1->stptr,255);
+   ipi[t1->stlen]=0;
+   DEREF(t1);
+   ret=ipi_check(ipi);
+	return make_number((AWKNUM)(ret));
+}
+
+/*  do_ipi_gen: Strukturierten Verwendungszweck (International Payment Instruction, IPI) testen */
+NODE *
+do_ipi_gen(int nargs)
+{
+	NODE *t1,*t2;
+   char zweck[256],out[64];
+   int ret;
+
+   t1=POP_SCALAR();
+   force_string(t1);
+   if(t1->stlen>255){
+      DEREF(t1);
+      return make_string("",0);
+   }
+   strncpy(zweck,t1->stptr,255);
+   zweck[t1->stlen]=0;
+   DEREF(t1);
+   ret=ipi_gen(zweck,out,NULL);
+	return make_string(out,strlen(out));
+}
+
+/*  do_iban_check: IBAN testen */
+NODE *
+do_iban_check(int nargs)
+{
+	NODE *t1,*t2;
+   char iban[256];
+   int ret;
+
+   t1=POP_SCALAR();
+   force_string(t1);
+   if(t1->stlen>255){
+      DEREF(t1);
+      return make_string("",0);
+   }
+   strncpy(iban,t1->stptr,255);
+   iban[t1->stlen]=0;
+   DEREF(t1);
+   lut_init(NULL,0,0);  /* falls noch nicht gemacht, initialisieren */
+   ret=iban_check(iban,NULL);
+	return make_number((AWKNUM)(ret));
+}
+
+/*  do_konto_check: Kontonummer testen */
+NODE *
+do_konto_check(int nargs)
+{
+	NODE *t1,*t2;
+   char blz[16],kto[16];
+   int ret;
+
+   POP_TWO_SCALARS(t1,t2);
+   force_string(t1);
+   force_string(t2);
+   if(t1->stlen>8){
+      DEREF(t1);
+      DEREF(t2);
+      return make_string("",0);
+   }
+   if(t2->stlen>15){
+      DEREF(t1);
+      DEREF(t2);
+      return make_string("",0);
+   }
+   strncpy(blz,t1->stptr,15);
+   blz[t1->stlen]=0;
+   strncpy(kto,t2->stptr,15);
+   kto[t2->stlen]=0;
+   DEREF(t1);
+   DEREF(t2);
+   ret=kto_check(blz,kto,NULL);
+	return make_number((AWKNUM)(ret));
+}
+
+/*  do_bank_valid: testen ob eine BLZ gültig ist
+ *  Die LUT-Routinen für den AWK-Port arbeiten nur mit den Hauptstellen,
+ *  der Parameter für die Nebenstellen wird immer auf 0 gesetzt.
+ */
+NODE *
+do_bank_valid(int nargs)
+{
+	NODE *t1;
+   char blz[16];
+   int ret;
+
+	t1=POP_SCALAR();
+	force_string(t1);
+   if(t1->stlen>8){
+      DEREF(t1);
+      return make_number((AWKNUM)(INVALID_BLZ_LENGTH));
+   }
+   strncpy(blz,t1->stptr,t1->stlen);
+   blz[t1->stlen]=0;
+	DEREF(t1);
+   ret=lut_blz(blz,0);
+   if(ret==LUT2_BLZ_NOT_INITIALIZED){
+      lut_init(NULL,7,0);
+      ret=lut_blz(blz,0);
+   }
+	return make_number((AWKNUM)(ret));
+}
+
+/*  do_bank_name: Banknamen zu einer BLZ bestimmen */
+NODE *
+do_bank_name(int nargs)
+{
+	NODE *t1;
+   char blz[16];
+   const char *name;
+   int pz,ret;
+
+	t1=POP_SCALAR();
+	force_string(t1);
+   if(t1->stlen>8){
+      DEREF(t1);
+      return make_number((AWKNUM)(INVALID_BLZ_LENGTH));
+   }
+   strncpy(blz,t1->stptr,t1->stlen);
+   blz[t1->stlen]=0;
+	DEREF(t1);
+   name=lut_name(blz,0,&ret);
+   if(ret==LUT2_NAME_NOT_INITIALIZED){
+      lut_init(NULL,7,0);
+      name=lut_name(blz,0,NULL);
+   }
+	return make_string(name,strlen(name));
+}
+
+/*  do_bank_name_kurz: Kurzbezeichnung der Bank zu einer BLZ bestimmen */
+NODE *
+do_bank_name_kurz(int nargs)
+{
+	NODE *t1;
+   char blz[16];
+   const char *name;
+   int pz,ret;
+
+	t1=POP_SCALAR();
+	force_string(t1);
+   if(t1->stlen>8){
+      DEREF(t1);
+      return make_number((AWKNUM)(INVALID_BLZ_LENGTH));
+   }
+   strncpy(blz,t1->stptr,t1->stlen);
+   blz[t1->stlen]=0;
+   name=lut_name_kurz(blz,0,&ret);
+   if(ret==LUT2_NAME_KURZ_NOT_INITIALIZED){
+      lut_init(NULL,7,0);
+      name=lut_name_kurz(blz,0,NULL);
+   }
+	DEREF(t1);
+	return make_string(name,strlen(name));
+}
+
+/*  do_bank_bic: BIC (Bank Identifier Code) zu einer BLZ bestimmen */
+NODE *
+do_bank_bic(int nargs)
+{
+	NODE *t1;
+   char blz[16];
+   const char *bic;
+   int pz,ret;
+
+	t1=POP_SCALAR();
+	force_string(t1);
+   if(t1->stlen>8){
+      DEREF(t1);
+      return make_number((AWKNUM)(INVALID_BLZ_LENGTH));
+   }
+   strncpy(blz,t1->stptr,t1->stlen);
+   blz[t1->stlen]=0;
+   bic=lut_bic(blz,0,&ret);
+   if(ret==LUT2_BIC_NOT_INITIALIZED){
+      lut_init(NULL,7,0);
+      bic=lut_bic(blz,0,NULL);
+   }
+	DEREF(t1);
+	return make_string(bic,strlen(bic));
+}
+
+/*  do_bank_plz: PLZ der Hauptstelle einer Bank suchen. */
+NODE *
+do_bank_plz(int nargs)
+{
+	NODE *t1;
+   char blz[16];
+   int plz,ret;
+
+	t1=POP_SCALAR();
+	force_string(t1);
+   if(t1->stlen>8){
+      DEREF(t1);
+      return make_number((AWKNUM)(INVALID_BLZ_LENGTH));
+   }
+   strncpy(blz,t1->stptr,t1->stlen);
+   blz[t1->stlen]=0;
+   plz=lut_plz(blz,0,&ret);
+   if(ret==LUT2_PLZ_NOT_INITIALIZED){
+      lut_init(NULL,7,0);
+      plz=lut_plz(blz,0,NULL);
+   }
+	DEREF(t1);
+	return make_number((AWKNUM)(plz));
+}
+
+/*  do_bank_ort: Ort des Hauptsitzes einer Bank bestimmen */
+NODE *
+do_bank_ort(int nargs)
+{
+	NODE *t1;
+   char blz[16];
+   const char *ort;
+   int pz,ret;
+
+	t1=POP_SCALAR();
+	force_string(t1);
+   if(t1->stlen>8){
+      DEREF(t1);
+      return make_number((AWKNUM)(INVALID_BLZ_LENGTH));
+   }
+   strncpy(blz,t1->stptr,t1->stlen);
+   blz[t1->stlen]=0;
+   ort=lut_ort(blz,0,&ret);
+   if(ret==LUT2_ORT_NOT_INITIALIZED){
+      lut_init(NULL,7,0);
+      ort=lut_ort(blz,0,NULL);
+   }
+	DEREF(t1);
+	return make_string(ort,strlen(ort));
+}
+
+/*  do_bank_pz: Prüfzifferberechnungsmethode für eine BLZ bestimmen */
+NODE *
+do_bank_pz(int nargs)
+{
+	NODE *t1;
+   char blz[16];
+   int pz,ret;
+
+	t1=POP_SCALAR();
+	force_string(t1);
+   if(t1->stlen>8){
+      DEREF(t1);
+      return make_number((AWKNUM)(INVALID_BLZ_LENGTH));
+   }
+   strncpy(blz,t1->stptr,t1->stlen);
+   blz[t1->stlen]=0;
+   pz=lut_pz(blz,0,&ret);
+   if(ret==LUT2_PZ_NOT_INITIALIZED){
+      lut_init(NULL,7,0);
+      pz=lut_pz(blz,0,NULL);
+   }
+	DEREF(t1);
+	return make_number((AWKNUM)(pz));
+}
+
+/*  kto_check_encoding: Kodierung festlegen */
+NODE *
+do_kto_check_encoding(int nargs)
+{
+	NODE *t1;
+   int mode,ret;
+
+   if(nargs){
+      t1=POP_SCALAR();
+      mode=(int)force_number(t1);
+      DEREF(t1);
+   }
+   else
+      mode=0;
+   ret=kto_check_encoding(mode);
+	return make_number((AWKNUM)(ret));
+}
+
+/*  do_retval2txt: Rückgabewert in Text umwandeln */
+NODE *
+do_retval2txt(int nargs)
+{
+	NODE *t1;
+   const char *ret_txt;
+
+	t1=POP_SCALAR();
+   ret_txt=kto_check_retval2txt((int)force_number(t1));
+   DEREF(t1);
+	return make_string(ret_txt,strlen(ret_txt));
+}
+
+/*  do_retval2utf8: Rückgabewert nach UTF-8 umwandeln */
+NODE *
+do_retval2utf8(int nargs)
+{
+	NODE *t1;
+   const char *ret_txt;
+
+	t1=POP_SCALAR();
+   ret_txt=kto_check_retval2utf8((int)force_number(t1));
+   DEREF(t1);
+	return make_string(ret_txt,strlen(ret_txt));
+}
diff -Nur gawk-4.0.1/Makefile.am gawk-4.0.1.neu/Makefile.am
--- gawk-4.0.1/Makefile.am	2012-03-28 21:47:53.000000000 +0200
+++ gawk-4.0.1.neu/Makefile.am	2012-05-31 23:22:23.143293448 +0200
@@ -23,6 +23,9 @@
 
 ## process this file with automake to produce Makefile.in
 
+# Make .bz2 files also.
+AUTOMAKE_OPTIONS = dist-bzip2
+
 # This variable insures that aclocal runs
 # correctly after changing configure.ac
 ACLOCAL_AMFLAGS = -I m4
@@ -52,7 +55,6 @@
 	m4 \
 	missing \
 	missing_d \
-	po/README \
 	pc \
 	posix \
 	regcomp.c \
@@ -63,11 +65,6 @@
 	vms \
 	ylwrap
 
-# It's OK for the generated file `version.c' not to be removed by
-# "make distclean".
-distcleancheck_listfiles = \
-  find . -type f -print | grep -v '^\./version\.c$$'
-
 # The order to do things in.
 # Build explicitly in "." in order to build gawk first, so
 # that `make check' without a prior `make' works.
@@ -101,6 +98,8 @@
 	getopt_int.h \
 	gettext.h \
 	io.c \
+	konto_check.c \
+	konto_check_awk.c \
 	mbsupport.h \
 	main.c \
 	msg.c \
@@ -120,7 +119,7 @@
 dgawk_SOURCES = $(base_sources) eval_d.c profile.c cmd.h command.y debug.c
 
 # Get extra libs as needed, Automake will supply LIBINTL and SOCKET_LIBS.
-LDADD = $(LIBSIGSEGV) $(LIBINTL) $(SOCKET_LIBS)
+LDADD = $(LIBSIGSEGV) $(LIBINTL) $(SOCKET_LIBS) -lz
 dgawk_LDADD = $(LDADD) @LIBREADLINE@
 
 # Directory for gawk's data files. Automake supplies datadir.
@@ -165,14 +164,8 @@
 check-local: gawk$(EXEEXT) pgawk$(EXEEXT)
 
 # A little extra clean up when making distributions.
-# And additional set up for the pc directory.
 dist-hook:
 	cd $(distdir)/extension ; rm -f *.o *.so
-	cd $(srcdir)/pc ; \
-	sed -n -f configpk.sed < ../configure.ac > tmp.sed ; \
-	sed -f config.sed < ../configh.in > config.tmp ; \
-	sed -f tmp.sed < config.tmp > config.h ; \
-	$(RM) tmp.sed config.tmp
 
 # Special rules for individual files
 # Use of awk instead of $(AWK) is deliberate, in case gawk doesn't build
diff -Nur gawk-4.0.1/Makefile.am.shared gawk-4.0.1.neu/Makefile.am.shared
--- gawk-4.0.1/Makefile.am.shared	1970-01-01 01:00:00.000000000 +0100
+++ gawk-4.0.1.neu/Makefile.am.shared	2012-05-31 22:09:42.784061519 +0200
@@ -0,0 +1,196 @@
+#
+# Makefile.am --- automake input file for gawk
+#
+# Copyright (C) 2000-2011 the Free Software Foundation, Inc.
+#
+# This file is part of GAWK, the GNU implementation of the
+# AWK Programming Language.
+#
+# GAWK is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# GAWK is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+#
+
+## process this file with automake to produce Makefile.in
+
+# Make .bz2 files also.
+AUTOMAKE_OPTIONS = dist-bzip2
+
+# This variable insures that aclocal runs
+# correctly after changing configure.ac
+ACLOCAL_AMFLAGS = -I m4
+
+# This insures that make flags get passed down to child makes.
+AM_MAKEFLAGS = 'CFLAGS=$(CFLAGS)' 'LDFLAGS=$(LDFLAGS)'
+
+# Stuff to include in the dist that doesn't need it's own
+# Makefile.am files
+EXTRA_DIST = \
+	ChangeLog.0 \
+	COPYING \
+	FUTURES \
+	INSTALL \
+	LIMITATIONS \
+	NEWS \
+	NEWS.0 \
+	POSIX.STD \
+	PROBLEMS \
+	README_d \
+	bisonfix.awk \
+	config.guess \
+	config.rpath  \
+	config.sub \
+	depcomp \
+	extension \
+	m4 \
+	missing \
+	missing_d \
+	pc \
+	posix \
+	regcomp.c \
+	regex_internal.c \
+	regex_internal.h \
+	regexec.c \
+	version.in \
+	vms \
+	ylwrap
+
+# The order to do things in.
+# Build explicitly in "." in order to build gawk first, so
+# that `make check' without a prior `make' works.
+SUBDIRS = \
+	. \
+	awklib \
+	doc \
+	po \
+	test
+
+# what to make and install
+bin_PROGRAMS = gawk pgawk dgawk
+
+# sources for both gawk and pgawk
+base_sources = \
+	array.c \
+	awk.h \
+	awkgram.y \
+	builtin.c \
+	custom.h \
+	dfa.c \
+	dfa.h \
+	ext.c \
+	field.c \
+	floatcomp.c \
+	floatmagic.h \
+	gawkmisc.c \
+	getopt.c \
+	getopt.h \
+	getopt1.c \
+	getopt_int.h \
+	gettext.h \
+	io.c \
+	konto_check_awk.c \
+	mbsupport.h \
+	main.c \
+	msg.c \
+	node.c \
+	protos.h \
+	random.c \
+	random.h \
+	re.c \
+	regex.c \
+	regex.h \
+	replace.c \
+	version.c \
+	xalloc.h
+
+gawk_SOURCES = $(base_sources) eval.c profile.c
+pgawk_SOURCES = $(base_sources) eval_p.c profile_p.c
+dgawk_SOURCES = $(base_sources) eval_d.c profile.c cmd.h command.y debug.c
+
+# Get extra libs as needed, Automake will supply LIBINTL and SOCKET_LIBS.
+LDADD = $(LIBSIGSEGV) $(LIBINTL) $(SOCKET_LIBS) -lkonto_check -lz
+dgawk_LDADD = $(LDADD) @LIBREADLINE@
+
+# Directory for gawk's data files. Automake supplies datadir.
+pkgdatadir = $(datadir)/awk
+
+# stuff for compiling gawk/pgawk
+DEFPATH='".$(PATH_SEPARATOR)$(pkgdatadir)"'
+
+DEFS= -DDEFPATH=$(DEFPATH) -DHAVE_CONFIG_H -DGAWK -DLOCALEDIR='"$(datadir)/locale"'
+
+# Get rid of core files when cleaning
+CLEANFILES = core core.*
+
+MAINTAINERCLEANFILES = version.c
+
+# We want hard links for install-exec-hook, below
+LN= ln
+
+# First, add a link from gawk to gawk-X.Y.Z.
+# Same for pgawk.
+#
+# For GNU systems where gawk is awk, add a link to awk.
+# (This is done universally, which may not always be right, but
+# there's no easy way to distinguish GNU from non-GNU systems.)
+install-exec-hook:
+	(cd $(DESTDIR)$(bindir); \
+	$(LN) gawk$(EXEEXT) gawk-$(VERSION)$(EXEEXT) 2>/dev/null ; \
+	$(LN) pgawk$(EXEEXT) pgawk-$(VERSION)$(EXEEXT) 2>/dev/null ; \
+	if [ ! -f awk ]; \
+	then	$(LN_S) gawk$(EXEEXT) awk; \
+	fi; exit 0)
+
+# Undo the above when uninstalling
+uninstall-links:
+	(cd $(DESTDIR)$(bindir); \
+	if [ -f awk ] && cmp awk gawk$(EXEEXT) > /dev/null; then rm -f awk; fi ; \
+	rm -f gawk-$(VERSION)$(EXEEXT) pgawk-$(VERSION)$(EXEEXT); exit 0)
+
+uninstall-recursive: uninstall-links
+
+# force there to be a gawk executable before running tests
+check-local: gawk$(EXEEXT) pgawk$(EXEEXT)
+
+# A little extra clean up when making distributions.
+dist-hook:
+	cd $(distdir)/extension ; rm -f *.o *.so
+
+# Special rules for individual files
+# Use of awk instead of $(AWK) is deliberate, in case gawk doesn't build
+# or work correctly.
+awkgram.c: awkgram.y
+	$(YACC) $(AM_YFLAGS) $(YFLAGS) $<
+	sed 's/parse error/syntax error/g' < y.tab.c | awk -f $(srcdir)/bisonfix.awk awkgram > $*.c && rm y.tab.c
+	if test -f y.tab.h; then \
+	if cmp -s y.tab.h $*.h; then rm -f y.tab.h; else mv y.tab.h $*.h; fi; \
+	else :; fi
+
+version.c: config.status version.in
+	$(SHELL) ./config.status --file=version.c:version.in
+
+command.c: command.y
+	$(YACC) -p zz $<
+	sed 's/parse error/syntax error/g' < y.tab.c | awk -f $(srcdir)/bisonfix.awk command > $*.c && rm y.tab.c
+
+# This is for my development & testing.
+efence: gawk
+	$(CC) $(LDFLAGS) -o gawk $$(ls *.o | grep -v '_p.o$$') $(LIBS) -lefence
+
+diffout valgrind-scan:
+	@cd test && $(MAKE) $(AM_MAKEFLAGS) $@
+
+valgrind:
+	cd test; rm -f log.[0-9]*; \
+	make check AWK="valgrind --leak-check=full --log-file=log.%p ../gawk"; \
+	make valgrind-scan
diff -Nur gawk-4.0.1/Makefile.am.static gawk-4.0.1.neu/Makefile.am.static
--- gawk-4.0.1/Makefile.am.static	1970-01-01 01:00:00.000000000 +0100
+++ gawk-4.0.1.neu/Makefile.am.static	2012-05-31 22:09:42.786061519 +0200
@@ -0,0 +1,197 @@
+#
+# Makefile.am --- automake input file for gawk
+#
+# Copyright (C) 2000-2011 the Free Software Foundation, Inc.
+#
+# This file is part of GAWK, the GNU implementation of the
+# AWK Programming Language.
+#
+# GAWK is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# GAWK is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+#
+
+## process this file with automake to produce Makefile.in
+
+# Make .bz2 files also.
+AUTOMAKE_OPTIONS = dist-bzip2
+
+# This variable insures that aclocal runs
+# correctly after changing configure.ac
+ACLOCAL_AMFLAGS = -I m4
+
+# This insures that make flags get passed down to child makes.
+AM_MAKEFLAGS = 'CFLAGS=$(CFLAGS)' 'LDFLAGS=$(LDFLAGS)'
+
+# Stuff to include in the dist that doesn't need it's own
+# Makefile.am files
+EXTRA_DIST = \
+	ChangeLog.0 \
+	COPYING \
+	FUTURES \
+	INSTALL \
+	LIMITATIONS \
+	NEWS \
+	NEWS.0 \
+	POSIX.STD \
+	PROBLEMS \
+	README_d \
+	bisonfix.awk \
+	config.guess \
+	config.rpath  \
+	config.sub \
+	depcomp \
+	extension \
+	m4 \
+	missing \
+	missing_d \
+	pc \
+	posix \
+	regcomp.c \
+	regex_internal.c \
+	regex_internal.h \
+	regexec.c \
+	version.in \
+	vms \
+	ylwrap
+
+# The order to do things in.
+# Build explicitly in "." in order to build gawk first, so
+# that `make check' without a prior `make' works.
+SUBDIRS = \
+	. \
+	awklib \
+	doc \
+	po \
+	test
+
+# what to make and install
+bin_PROGRAMS = gawk pgawk dgawk
+
+# sources for both gawk and pgawk
+base_sources = \
+	array.c \
+	awk.h \
+	awkgram.y \
+	builtin.c \
+	custom.h \
+	dfa.c \
+	dfa.h \
+	ext.c \
+	field.c \
+	floatcomp.c \
+	floatmagic.h \
+	gawkmisc.c \
+	getopt.c \
+	getopt.h \
+	getopt1.c \
+	getopt_int.h \
+	gettext.h \
+	io.c \
+	konto_check.c \
+	konto_check_awk.c \
+	mbsupport.h \
+	main.c \
+	msg.c \
+	node.c \
+	protos.h \
+	random.c \
+	random.h \
+	re.c \
+	regex.c \
+	regex.h \
+	replace.c \
+	version.c \
+	xalloc.h
+
+gawk_SOURCES = $(base_sources) eval.c profile.c
+pgawk_SOURCES = $(base_sources) eval_p.c profile_p.c
+dgawk_SOURCES = $(base_sources) eval_d.c profile.c cmd.h command.y debug.c
+
+# Get extra libs as needed, Automake will supply LIBINTL and SOCKET_LIBS.
+LDADD = $(LIBSIGSEGV) $(LIBINTL) $(SOCKET_LIBS) -lz
+dgawk_LDADD = $(LDADD) @LIBREADLINE@
+
+# Directory for gawk's data files. Automake supplies datadir.
+pkgdatadir = $(datadir)/awk
+
+# stuff for compiling gawk/pgawk
+DEFPATH='".$(PATH_SEPARATOR)$(pkgdatadir)"'
+
+DEFS= -DDEFPATH=$(DEFPATH) -DHAVE_CONFIG_H -DGAWK -DLOCALEDIR='"$(datadir)/locale"'
+
+# Get rid of core files when cleaning
+CLEANFILES = core core.*
+
+MAINTAINERCLEANFILES = version.c
+
+# We want hard links for install-exec-hook, below
+LN= ln
+
+# First, add a link from gawk to gawk-X.Y.Z.
+# Same for pgawk.
+#
+# For GNU systems where gawk is awk, add a link to awk.
+# (This is done universally, which may not always be right, but
+# there's no easy way to distinguish GNU from non-GNU systems.)
+install-exec-hook:
+	(cd $(DESTDIR)$(bindir); \
+	$(LN) gawk$(EXEEXT) gawk-$(VERSION)$(EXEEXT) 2>/dev/null ; \
+	$(LN) pgawk$(EXEEXT) pgawk-$(VERSION)$(EXEEXT) 2>/dev/null ; \
+	if [ ! -f awk ]; \
+	then	$(LN_S) gawk$(EXEEXT) awk; \
+	fi; exit 0)
+
+# Undo the above when uninstalling
+uninstall-links:
+	(cd $(DESTDIR)$(bindir); \
+	if [ -f awk ] && cmp awk gawk$(EXEEXT) > /dev/null; then rm -f awk; fi ; \
+	rm -f gawk-$(VERSION)$(EXEEXT) pgawk-$(VERSION)$(EXEEXT); exit 0)
+
+uninstall-recursive: uninstall-links
+
+# force there to be a gawk executable before running tests
+check-local: gawk$(EXEEXT) pgawk$(EXEEXT)
+
+# A little extra clean up when making distributions.
+dist-hook:
+	cd $(distdir)/extension ; rm -f *.o *.so
+
+# Special rules for individual files
+# Use of awk instead of $(AWK) is deliberate, in case gawk doesn't build
+# or work correctly.
+awkgram.c: awkgram.y
+	$(YACC) $(AM_YFLAGS) $(YFLAGS) $<
+	sed 's/parse error/syntax error/g' < y.tab.c | awk -f $(srcdir)/bisonfix.awk awkgram > $*.c && rm y.tab.c
+	if test -f y.tab.h; then \
+	if cmp -s y.tab.h $*.h; then rm -f y.tab.h; else mv y.tab.h $*.h; fi; \
+	else :; fi
+
+version.c: config.status version.in
+	$(SHELL) ./config.status --file=version.c:version.in
+
+command.c: command.y
+	$(YACC) -p zz $<
+	sed 's/parse error/syntax error/g' < y.tab.c | awk -f $(srcdir)/bisonfix.awk command > $*.c && rm y.tab.c
+
+# This is for my development & testing.
+efence: gawk
+	$(CC) $(LDFLAGS) -o gawk $$(ls *.o | grep -v '_p.o$$') $(LIBS) -lefence
+
+diffout valgrind-scan:
+	@cd test && $(MAKE) $(AM_MAKEFLAGS) $@
+
+valgrind:
+	cd test; rm -f log.[0-9]*; \
+	make check AWK="valgrind --leak-check=full --log-file=log.%p ../gawk"; \
+	make valgrind-scan
diff -Nur gawk-4.0.1/Makefile.in gawk-4.0.1.neu/Makefile.in
--- gawk-4.0.1/Makefile.in	2012-03-28 22:03:23.000000000 +0200
+++ gawk-4.0.1.neu/Makefile.in	2012-05-31 23:22:28.583543917 +0200
@@ -90,7 +90,8 @@
 am__objects_1 = array.$(OBJEXT) awkgram.$(OBJEXT) builtin.$(OBJEXT) \
 	dfa.$(OBJEXT) ext.$(OBJEXT) field.$(OBJEXT) \
 	floatcomp.$(OBJEXT) gawkmisc.$(OBJEXT) getopt.$(OBJEXT) \
-	getopt1.$(OBJEXT) io.$(OBJEXT) main.$(OBJEXT) msg.$(OBJEXT) \
+	getopt1.$(OBJEXT) io.$(OBJEXT) konto_check.$(OBJEXT) \
+	konto_check_awk.$(OBJEXT) main.$(OBJEXT) msg.$(OBJEXT) \
 	node.$(OBJEXT) random.$(OBJEXT) re.$(OBJEXT) regex.$(OBJEXT) \
 	replace.$(OBJEXT) version.$(OBJEXT)
 am_dgawk_OBJECTS = $(am__objects_1) eval_d.$(OBJEXT) profile.$(OBJEXT) \
@@ -170,9 +171,10 @@
     dir1=`echo "$$dir1" | sed -e "$$sed_rest"`; \
   done; \
   reldir="$$dir2"
-DIST_ARCHIVES = $(distdir).tar.gz $(distdir).tar.xz
+DIST_ARCHIVES = $(distdir).tar.gz $(distdir).tar.bz2 $(distdir).tar.xz
 GZIP_ENV = --best
 distuninstallcheck_listfiles = find . -type f -print
+distcleancheck_listfiles = find . -type f -print
 
 # Directory for gawk's data files. Automake supplies datadir.
 pkgdatadir = $(datadir)/awk
@@ -297,6 +299,9 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 
+# Make .bz2 files also.
+AUTOMAKE_OPTIONS = dist-bzip2
+
 # This variable insures that aclocal runs
 # correctly after changing configure.ac
 ACLOCAL_AMFLAGS = -I m4
@@ -326,7 +331,6 @@
 	m4 \
 	missing \
 	missing_d \
-	po/README \
 	pc \
 	posix \
 	regcomp.c \
@@ -338,12 +342,6 @@
 	ylwrap
 
 
-# It's OK for the generated file `version.c' not to be removed by
-# "make distclean".
-distcleancheck_listfiles = \
-  find . -type f -print | grep -v '^\./version\.c$$'
-
-
 # The order to do things in.
 # Build explicitly in "." in order to build gawk first, so
 # that `make check' without a prior `make' works.
@@ -375,6 +373,8 @@
 	getopt_int.h \
 	gettext.h \
 	io.c \
+	konto_check.c \
+	konto_check_awk.c \
 	mbsupport.h \
 	main.c \
 	msg.c \
@@ -394,7 +394,7 @@
 dgawk_SOURCES = $(base_sources) eval_d.c profile.c cmd.h command.y debug.c
 
 # Get extra libs as needed, Automake will supply LIBINTL and SOCKET_LIBS.
-LDADD = $(LIBSIGSEGV) $(LIBINTL) $(SOCKET_LIBS)
+LDADD = $(LIBSIGSEGV) $(LIBINTL) $(SOCKET_LIBS) -lz
 dgawk_LDADD = $(LDADD) @LIBREADLINE@
 
 # stuff for compiling gawk/pgawk
@@ -531,6 +531,8 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/getopt.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/getopt1.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/io.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/konto_check.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/konto_check_awk.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/main.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/msg.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/node.Po@am__quote@
@@ -767,7 +769,6 @@
 dist-gzip: distdir
 	tardir=$(distdir) && $(am__tar) | GZIP=$(GZIP_ENV) gzip -c >$(distdir).tar.gz
 	$(am__remove_distdir)
-
 dist-bzip2: distdir
 	tardir=$(distdir) && $(am__tar) | bzip2 -9 -c >$(distdir).tar.bz2
 	$(am__remove_distdir)
@@ -794,6 +795,7 @@
 
 dist dist-all: distdir
 	tardir=$(distdir) && $(am__tar) | GZIP=$(GZIP_ENV) gzip -c >$(distdir).tar.gz
+	tardir=$(distdir) && $(am__tar) | bzip2 -9 -c >$(distdir).tar.bz2
 	tardir=$(distdir) && $(am__tar) | xz -c >$(distdir).tar.xz
 	$(am__remove_distdir)
 
@@ -1032,14 +1034,8 @@
 check-local: gawk$(EXEEXT) pgawk$(EXEEXT)
 
 # A little extra clean up when making distributions.
-# And additional set up for the pc directory.
 dist-hook:
 	cd $(distdir)/extension ; rm -f *.o *.so
-	cd $(srcdir)/pc ; \
-	sed -n -f configpk.sed < ../configure.ac > tmp.sed ; \
-	sed -f config.sed < ../configh.in > config.tmp ; \
-	sed -f tmp.sed < config.tmp > config.h ; \
-	$(RM) tmp.sed config.tmp
 
 # Special rules for individual files
 # Use of awk instead of $(AWK) is deliberate, in case gawk doesn't build
